src/aeso_ast_infer_types.erl

```diff
@@ -862,7 +867,9 @@ infer_letrec(Env, Defs) ->
     [print_typesig(S) || S <- TypeSigs],
     {TypeSigs, NewDefs}.
 
-infer_letfun(Env, {letfun, Attrib, Fun = {id, NameAttrib, Name}, Args, What, Body}) ->
+infer_letfun(Env0, {letfun, Attrib, Fun = {id, NameAttrib, Name}, Args, What, Body}) ->
+    Env = Env0#env{ stateful = aeso_syntax:get_ann(stateful, Attrib, false),
```

Comment:
does this really work? Attribute must then contain {stateful, true}, I guess.

Reply:
```erlang
1> proplists:get_value(foo, [foo]).                                                                       
true
```